sdks/go/alpha.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2020 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sdk
+
+import (
+	"context"
+
+	"agones.dev/agones/pkg/sdk/alpha"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+)
+
+// Alpha is the struct for Alpha SDK functionality
+type Alpha struct {
+	client alpha.SDKClient
+	ctx    context.Context
+}
+
+// newAlpha creates a new Alpha SDK with the passed in connection.
+func newAlpha(conn *grpc.ClientConn) *Alpha {
+	return &Alpha{
+		ctx:    context.Background(),
```

Comment:
Maybe this is how all of the existing code works, but it's more typical to pass context in with each rpc rather than store it in a struct and reuse it. I know from experience that it can be quite a pain to retrofit code to pass context down the stack later...

Reply:
Hmnn, that's an interesting point. I'm also not sure why we're storing the background context, since it's already stored inside the Context package.

Any issue to shifting the alpha implementations to look like:

```go
// GetPlayerCapacity gets the last player capacity that was set through the SDK.
// if the player capacity is set from outside the SDK, use SDK.Alpha.GameServer() instead.
func (a *Alpha) GetPlayerCapacity() (int64, error) {
	c, err := a.client.GetPlayerCapacity(context.Background(), &alpha.Empty{})
	return c.Count, errors.Wrap(err, "could not get player capacity")
}
```

Since we're never attempting to cancel them?

I figure the others could be cleaned up in a later PR as needed?

Reply:
Context isn't just for canceling though; it can also be used to pass things like authn/authz tokens, trace ids (to track call dependencies), etc. 

I think the suggested change is better than what you currently have, but is there a reason that (for golang clients) we don't set the function signature to include the context? e.g.

```
// GetPlayerCapacity gets the last player capacity that was set through the SDK.
// if the player capacity is set from outside the SDK, use SDK.Alpha.GameServer() instead.
func (a *Alpha) GetPlayerCapacity(ctx context.Context) (int64, error) {
	c, err := a.client.GetPlayerCapacity(ctx, &alpha.Empty{})
	return c.Count, errors.Wrap(err, "could not get player capacity")
}
```

Reply:
Probably that it was just how it was done originally, nothing more than that, and to keep it inline with the other SDKs.

I'll make the change to move these parts to use the background Context.